spaceroombas/server.py

```diff
@@ -0,0 +1,226 @@
+
+@app.route("/")
+def hello_world():
+    return ""
+
+@app.route('/api/v1', methods=["GET"])
+def info_view():
+    """List of routes for this API."""
+    output = {
+            "Server": "placeholder.class",
+            "IP": "IP.class"
+
+    }
+    return output
+    
+
+@app.route("/post", methods = ['POST'])
+def postJson():
+    try:
+        print (request.is_json)
+        content = request.get_json()
+        print(content)
+        print(content['device'])
+        print(content['value'])
+        print(content['timestamp'])
+        return content['timestamp']
+    except TypeError:
+        return ""
+
+@app.route("/register", methods= ["POST"])
+def register_user():
+    credentials = _safe_fetch_json_credentials(request.get_json())
+   
+    if len(credentials.username) > 15:
+        return("Username exceeds 15 characters"), 400
+    if len(credentials.password) > 20:
+        return("Password exceeds 20 characters"), 400
+    
+    newUser =  UserDbInfo(username=credentials.username, password=credentials.hashed_password, email=credentials.email)
+    
+    if not(credentials is None):
+        newUser = UserDbInfo.query.filter_by(email=credentials.email).first()
+        if newUser is None:
+            newUser =  UserDbInfo(username=credentials.username, password=credentials.hashed_password, email=credentials.email)
+            db.session.add(newUser)
+            try:
+                db.session.commit()
+            except IntegrityError:
+                return "Username already exists", 400
+            return jsonify({"status":"registered"})
+        else:
+            return("User already exists"), 400
+    return("Please insert a username and password"), 400
+    
+
+@app.route("/login", methods= ["POST"])
+def authenticate_route():
+    credentials = _safe_fetch_json_credentials(request.get_json())
+    
+    # TODO probably should return some sort of JSON error object here
+    if credentials.username is None:
+        return("Please insert your username"), 400
+    if credentials.password is None:
+        return("Please insert your password"), 400
+    
+    userLogin = UserDbInfo.query.filter_by(username=credentials.username).first()
+
+    if userLogin:
+        if userLogin.password == credentials.hashed_password:
+            token = jwt.encode({
+                'user':userLogin.id,
+                'exp' : datetime.datetime.utcnow() + datetime.timedelta(seconds=600)
+            },
+                app.config['SECRET_KEY'])
+            return jsonify({'token': token})
+        else:
+            return("Login Information is Not Valid"), 403
+    
+    return("Login Information is Not Valid"), 403
+
+@app.route("/auth", methods= ["GET"])
+@token_required
+def authorized():
+    return jsonify({
+        "message":"User is authorized."
+    })
+    
+@app.route("/forgotPass", methods = ["GET", "POST"]) #This method takes in the user email from Unity. Upon submit, user is sent email
```

Comment:
This most likely should only be a POST request

Reply:
changed to only post